docker/auth/auth.py

```diff
@@ -136,10 +136,19 @@ def load_config(config_path=None):
     """
     Loads authentication data from a Docker configuration file in the given
     root directory or if config_path is passed use given path.
+    Lookup priority:
+        explicit config_path parameter > DOCKER_CONFIG environment variable >
+        ~/.docker/config.json > ~/.dockercfg
     """
     conf = {}
     data = None
 
+    if not config_path and os.environ.get('DOCKER_CONFIG'):
+        config_path = os.path.join(
+            os.environ.get('DOCKER_CONFIG'),
+            os.path.basename(DOCKER_CONFIG_FILENAME)
+        )
+
     # Prefer ~/.docker/config.json.
     config_file = config_path or os.path.join(os.path.expanduser('~'),
                                               DOCKER_CONFIG_FILENAME)
```

Comment:
I'd like to make a suggestion here that I think will make it easier to test.  I think the logic for "find config file" can be separated from "read contents of config file".

```python

def find_config(config_path=None):
    for path in (
        config_path, 
        os.path.join(os.environ.get('DOCKER_CONFIG'), os.path.basename(DOCKER_CONFIG_FILENAME)),
        os.path.join(os.path.expanduser('~'), DOCKER_CONFIG_FILENAME),
        os.path.exists(os.path.join(os.path.expanduser('~'), LEGACY_DOCKER_CONFIG_FILENAME))
    ):
        if path and os.path.exists(path):
            return path

    return None
```